demos/random_cut_demo/src/main/scala/io/hydrosphere/rrcf/impl/RRCT.scala

```diff
@@ -34,7 +32,9 @@ class RRCT(initialData: Array[DataPoint], timeDecay: Int = 1000) {
     val dataPointBoundingBox = new BoundingBox(newDataPoint)
     val commonBoundingBox = dataPointBoundingBox.union(possibleSiblingNode.boundingBox)
     val diameters = commonBoundingBox.minValues zip commonBoundingBox.maxValues map { case (min, max) => max - min }
-    val diameterCut = diameters.sum * r.nextDouble() // TODO it samples from semi-interval [0;1) but we need [0;1]
+    var randomCoeff = 0.0
+    do randomCoeff = r.nextDouble() while (randomCoeff == 0.0) // check to not make cut at 0.0

```

Comment:
are there any benefits of using (0; 1) interval instead of [0; 1]?

Reply:
If you cut at 0 or 1 - your cut will divide bounding box into itself and empty space. Algorithmically this cut will be useless. Behaviour for such event is undefined.